src/pudl_archiver/archivers/phmsagas.py

```diff
@@ -76,15 +76,26 @@ async def get_zip_resource(
         if form not in PHMSA_FORMS:
             logger.warning(f"New form type found: {form}.")
 
+        download_path = self.download_directory / f"{self.name}-{filename}.zip"
+        await self.download_zipfile(url, download_path)
+
         # From start and end year, get partitions
         start_year = int(filename.split("_")[-2])
         end_year = filename.split("_")[-1]
-        # If end year is present, this will include data through last year.
-        end_year = datetime.now().year - 1 if end_year == "present" else int(end_year)
-        years = list(range(start_year, end_year + 1))
 
-        download_path = self.download_directory / f"{self.name}-{filename}.zip"
-        await self.download_zipfile(url, download_path)
+        # If end year is present, open zipfile and get last year of excel data in it.
+        if end_year == "present":
+            excel_files = sorted(
+                [
+                    file
+                    for file in ZipFile(download_path).namelist()
+                    if file.endswith(".xlsx")
+                ]
+            )
+            end_year = int(excel_files[-1].split("_")[-1].replace(".xlsx", ""))
```

Comment:
okay okay your sorting them so the last one is the one with the max year? grabbed via the -1? this is not a blocking concern at all but i think it would be more readable if you attempted to grab the year out of each file and `end_year` was just the max